base_comment_template/models/comment_template.py

```diff
@@ -20,9 +20,14 @@ class CommentTemplate(models.AbstractModel):
 
     comment_template_ids = fields.Many2many(
         compute="_compute_comment_template_ids",
+        compute_sudo=True,
         comodel_name="base.comment.template",
         string="Comment Template",
-        domain=lambda self: [("model_ids.model", "=", self._name)],
+        # model_ids.model is not allowed because the user probably does
+        # not have read access to ir.model
+        # Using like operator is not a problem because only models with
+        # is_comment_template=True flag can exist.
+        domain=lambda self: [("models", "like", self._name)],
```

Comment:
This is still incorrect, as it will match any model with a longer name, but including the model name.

Reply:
What do you think about this?
```

[
    ("models", "=", self._name),
    ("models", "=like", ",%s" % self._name),
    ("models", "=like", "%s," % self._name)
]
```

`sale.order` model (for example) could appear in the following ways:
- sale.order
- other.model,sale.order
- sale.order,other.model

Reply:
you have to put the `search` attribute on the computed field, and search splitting by comma.

Reply:
I'm sorry but I don't understand how to apply what you indicate in a domain.